### PR TITLE
Typemap: Set init to false after each individual destructor

### PR DESCRIPTION
**Pull request by Megamouse**
Fixes crashes when fxo destructors call other fxo.

### CODE REVIEW

**Comment by elad335** [comment]
Great catch!!

File: rpcs3/util/fixed_typemap.hpp [hunk 1]
```diff
@@ -299,21 +299,22 @@ namespace stx
 			}
 
 			// Get actual number of created objects
-			u32 _max = 0;
+			std::vector<usz> types_to_destroy;
```
Comment (elad335) [inline rpcs3/util/fixed_typemap.hpp:302]: Althogh this is probably not needed, there must be another way to get the typeinfo index.

Comment (Megamouse) [reply rpcs3/util/fixed_typemap.hpp:302]: it's wrong anyway, since it's the wrong order.

Comment (Megamouse) [reply rpcs3/util/fixed_typemap.hpp]: probably fixed now

File: rpcs3/util/fixed_typemap.hpp [hunk 2]
```diff
@@ -313,7 +313,12 @@ namespace stx
 			// Destroy objects in reverse order
 			for (; _max; _max--)
 			{
-				(*--m_info)->destroy(*--m_order);
+				auto* info = *--m_info;
+				const u32 type_index = static_cast<const type_info<typeinfo>*>(info)->index();
+				info->destroy(*--m_order);
+
+				// Set init to false. We don't want other fxo to use this fxo in their destructor.
+				m_init[type_index] = false;
```
Comment (elad335) [inline rpcs3/util/fixed_typemap.hpp:321]: Do it in info->destroy

Comment (Megamouse) [reply rpcs3/util/fixed_typemap.hpp:321]: It's not the same object?

Comment (elad335) [reply rpcs3/util/fixed_typemap.hpp:321]: I mean, move this line to typeinfo::destroy. It is the base class.

Comment (Megamouse) [reply rpcs3/util/fixed_typemap.hpp:321]: it's a static function...

Comment (elad335) [reply rpcs3/util/fixed_typemap.hpp:321]: Then pass it as argument or smth

Comment (elad335) [reply rpcs3/util/fixed_typemap.hpp:321]: And do it in auto_typemap.hpp too

Comment (Megamouse) [reply rpcs3/util/fixed_typemap.hpp:321]: Sorry but I don't have the time to make this code even more spaghetti-like

**Pull request merged**